test/erc1155m/ERC1155MInitializableTest.t.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {Test} from "forge-std/Test.sol";
+import {LibClone} from "solady/src/utils/LibClone.sol";
+
+import {ERC1155MInitializableV1_0_1 as ERC1155MInitializable} from
+    "../../contracts/nft/erc1155m/ERC1155MInitializableV1_0_1.sol";
+import {MintStageInfo1155} from "../../contracts/common/Structs.sol";
+import {ErrorsAndEvents} from "../../contracts/common/ErrorsAndEvents.sol";
+
+contract ERC1155MInitializableTest is Test {
```

Comment:
note that we are still lacking full test coverage with foundry for erc1155, though we have coverage with hardhat.

Reply:
these tests aim to add coverage for the `transferable` and setup lock features